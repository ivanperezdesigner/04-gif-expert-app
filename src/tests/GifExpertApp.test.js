import React from 'react';
import { shallow } from 'enzyme'
import GifExpertApp from '../GifExpertApp'

describe('Probando el componenre <GifExpertApp', () => {
    test('debe cargar correctamente el componenete', () => {
        const wrapper = shallow(<GifExpertApp />)
        expect(wrapper).toMatchSnapshot()
    })
    
    test('Debe mostrar una lista de categorÃ­as', () => {
        const categories = ['figth club', 'Amelie']
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />)
        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('GifGrid').length).toBe(categories.length)
    })
})
