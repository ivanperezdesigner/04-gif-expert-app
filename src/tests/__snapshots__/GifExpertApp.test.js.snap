// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando el componenre <GifExpertApp Debe mostrar una lista de categorÃ­as 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="figth club"
      key="figth club"
    />
    <GifGrid
      category="Amelie"
      key="Amelie"
    />
  </ol>
</Fragment>
`;

exports[`Probando el componenre <GifExpertApp debe cargar correctamente el componenete 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
